File: spec/php/RepeatEosStructTest.php
<?php
namespace Kaitai\Struct\Tests;

class RepeatEosStructTest extends TestCase {
    public function testRepeatEosStruct() {
        $r = RepeatEosStruct::fromFile(self::SRC_DIR_PATH . "/repeat_eos_struct.bin");

        $this->assertEquals(2, count($r->chunks));
        $this->assertEquals(0, $r->chunks[0]->offset);
        $this->assertEquals(0x42, $r->chunks[0]->len);
        $this->assertEquals(0x42, $r->chunks[1]->offset);
        $this->assertEquals(0x815, $r->chunks[1]->len);
    }
}

File: spec/go/repeat_eos_struct_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestRepeatEosStruct(t *testing.T) {
	f, err := os.Open("../../src/repeat_eos_struct.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r RepeatEosStruct
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, 2, len(r.Chunks))
	assert.EqualValues(t, 0, r.Chunks[0].Offset)
	assert.EqualValues(t, 0x42, r.Chunks[0].Len)
	assert.EqualValues(t, 0x42, r.Chunks[1].Offset)
	assert.EqualValues(t, 0x815, r.Chunks[1].Len)
}
